packages/bs-multichain/src/features/bridge/Neo3NeoXBridgeOrchestrator.ts

```diff
@@ -0,0 +1,475 @@
+    this.#internalTokenToReceive = {
+      ...this.#internalTokenToReceive,
+      ...value,
+    }
+    this.eventEmitter.emit('tokenToReceive', this.#internalTokenToReceive)
+  }
+
+  get #addressToReceive(): TBridgeValidateValue<string> {
+    return this.#internalAddressToReceive
+  }
+  set #addressToReceive(value: Partial<TBridgeValidateValue<string>>) {
+    this.#internalAddressToReceive = {
+      ...this.#internalAddressToReceive,
+      ...value,
+    }
+    this.eventEmitter.emit('addressToReceive', this.#internalAddressToReceive)
+  }
+
+  get #amountToReceive(): TBridgeValue<string> {
+    return this.#internalAmountToReceive
+  }
+  set #amountToReceive(value: Partial<TBridgeValue<string>>) {
+    this.#internalAmountToReceive = {
+      ...this.#internalAmountToReceive,
+      ...value,
+    }
+    this.eventEmitter.emit('amountToReceive', this.#internalAmountToReceive)
+  }
+
+  get #tokenToUseBalance(): TBridgeValue<BalanceResponse | undefined> {
+    return this.#internalTokenToUseBalance
+  }
+  set #tokenToUseBalance(value: Partial<TBridgeValue<BalanceResponse>>) {
+    this.#internalTokenToUseBalance = {
+      ...this.#internalTokenToUseBalance,
+      ...value,
+    }
+    this.eventEmitter.emit('tokenToUseBalance', this.#internalTokenToUseBalance)
+  }
+
+  get #bridgeFee(): TBridgeValue<string> {
+    return this.#internalBridgeFee
+  }
+  set #bridgeFee(value: Partial<TBridgeValue<string>>) {
+    this.#internalBridgeFee = {
+      ...this.#internalBridgeFee,
+      ...value,
+    }
+    this.eventEmitter.emit('bridgeFee', this.#internalBridgeFee)
+  }
+
+  #treatError(error: unknown): BSError {
+    if (error instanceof BSError) return error
+    return new BSError('An unexpected error occurred', 'UNEXPECTED_ERROR', { cause: error })
+  }
+
+  async init(): Promise<void> {
+    this.#availableTokensToUse = {
+      value: this.fromService.neo3NeoXBridgeService.tokens,
+    }
+
+    this.#accountToUse = { value: null, loading: false }
+    this.#addressToReceive = { value: null, valid: null, loading: false }
+    this.#amountToUse = { value: null, valid: null, loading: false }
+    this.#amountToUseMin = { value: null, loading: false }
+    this.#amountToUseMax = { value: null, loading: false }
+    this.#amountToReceive = { value: null, loading: false }
+    this.#tokenToUse = { value: null, loading: false }
+    this.#tokenToReceive = { value: null, loading: false }
+    this.#tokenToUseBalance = { value: null, loading: false }
+    this.#bridgeFee = { value: null, loading: false }
+    this.#transactionHash = null
+    this.#balances = null
+    this.#feeTokenBalance = null
+  }
+
+  async switchTokens(): Promise<void> {
+    const fromService = this.fromService
+    const toService = this.toService
+
+    this.fromService = toService
+    this.toService = fromService
+
+    const tokenToReceive = this.#tokenToReceive.value
+
+    await this.init()
+    await this.setTokenToUse(tokenToReceive)
```

Comment:
You could add the `await this.setTokenToReceive(tokenToUse)` here too, couldn't you?